app/src/main/java/ch/epfl/sweng/SDP/auth/Account.java

```diff
@@ -129,7 +132,9 @@ public static void createAccount(Context context, ConstantsWrapper constantsWrap
      */
     public static Account getInstance(Context context) {
         if (instance == null) {
-            createAccount(context, new ConstantsWrapper(), "", "");
+            ConstantsWrapper constantsWrapper = new ConstantsWrapper();
+            createAccount(context, constantsWrapper,
+                    constantsWrapper.getFirebaseUserId(), "");
```

Comment:
Maybe the problem related to empty usernames comes from here, just put « username » to be sure.